README.md

```diff
@@ -15,6 +15,12 @@ in a more efficient manner than performing each operation one at a time.
 
 For a mutable variant, see [go-radix](https://github.com/armon/go-radix).
 
+## :warning: This tree implementation is low-level and can be easily used incorrectly
+   - **No key may be a prefix of any other.**
+     - This is _not_ enforced but _will cause undefined behavior_ if violated. 
+     - The typical way to work around this is to append a null byte (`0x0`) to each key unless you know in advance your keys will never be prefixes. We don't always do this in tests if we arranged them to not be prefixes anyway!
+     - We could avoid this gotcha by automatically adding null bytes to all keys but that has high cost of causing re-allocation of keys for every operation that the caller might not want if they already.
```

Comment:
> if they already.

This sentence seems to end abruptly.

What about adding new methods e.g. `SafeInsert` that append the null byte?  So people have to opt-in, but at least new users of the lib are guided to using it in a way with less footguns?